tests/cli-misc.rs

```diff
@@ -110,6 +110,33 @@ fn upgrade_v2_metadata_to_v12() {
     });
 }
 
+// Verifies the conversion from separate files to a single settings.toml
+#[test]
+fn upgrade_toml_settings() {
+    setup(&|config| {
+        rustup_utils::raw::write_file(&config.rustupdir.join("version"), "2").unwrap();
+        rustup_utils::raw::write_file(&config.rustupdir.join("default"), "beta").unwrap();
+        rustup_utils::raw::write_file(&config.rustupdir.join("overrides"),
+                                      "a;nightly\nb;stable").unwrap();
+        rustup_utils::raw::write_file(&config.rustupdir.join("telemetry-on"), "").unwrap();
+        expect_err(config, &["rustup", "default", "nightly"],
+                   "rustup's metadata is out of date. run `rustup self upgrade-data`");
```

Comment:
Why is this expecting an error message saying to run `upgrade-data`? Doesn't this upgrade happen silently?

I think this test should be tweaked just a little. Before writing the old metadata, first run the commands that correspond to the metadata, then write the metadata and delete the settings.toml file. Then run some `rustup` command to force the silent upgrade. Then run further `rustc` commands that verify that the metadata is intact - that `rustc --version` prints the right thing for the default and override (telemetry flag probably not worth testing explicitly).